### ISSUE TITLE
nnp-comp2 usage

### ISSUE THREAD

**Issue by Matheus-Novoa** [CompPhysVienna/n2p2#158]
I had some doubts about the "Extrapolation problems" issue.

1. How I set the <ew_threshold>, <energy_threshold>, <force_threshold> variables?
2. 
Which does settings it make sense modify in NNP1 and NNP2?

3. In the example folder, the only differences between the two input.nn files were that in one of them some symmetry function lines were commented out and in the other they were not. Are these modifications enough to distinguish one NNP from another?

4. Are there other settings that can be changed?

**Comment by owenrett** [comment]
Disclaimer: not a developer, so answer here is just based off of experience using the tools. 

**1)How I set the <ew_threshold>, <energy_threshold>, <force_threshold> variables?**

You call the function nnp-comp2 by first running
```nnp-comp2 compare``` 

to get predictions from the two neural networks. Then you can specify the ew_threshold, energy_threshold, force_threshold parameters by calling 

```nnp-comp2 select <ew_threshold> <energy_threshold> <force_threshold> <species_name_1> <species_name_2> ...```

where ew_threshold is the max number of extrapolation warnings before selecting a structure, energy_threshold is the max energy difference between two structures (on a per-atom basis), force_threshold is the max difference between any two predicted forces, and species_name_1, species_name_2,.. are the elements present. 

**3) In the example folder, the only differences between the two input.nn files were that in one of them some symmetry function lines were commented out and in the other they were not. Are these modifications enough to distinguish one NNP from another?**

When using nnp-comp2 to generate new training data you want to keep all elements of the two networks the same **except** the ``random_seed``` to initialize the weights. This ensures that any differences in predictions between the two networks is due to this random initialization, not network structure differences. In this way, the structures where large differences exist, and are selected by nnp-comp2, are areas where the model is learning poorly. Any other changed parameters could mean that differences are not due to poor learning.

**Comment by Matheus-Novoa** [comment]
**"... ew_threshold is the max number of extrapolation warnings before selecting a structure ..."**
So, in this example the the ew_threshold would be 1374 ?
![image](https://user-images.githubusercontent.com/65021130/155154230-0f2028d0-a338-4985-8f60-2fb25e0b22b4.png)

**Comment by owenrett** [comment]
Generally the ew_threshold should be set to the maximum number of extrapolation warnings you think is acceptable in your potential (generally speaking this is 0), and it will automatically select any structure with more than that number of extrapolation warnings. 

The **energy_threshold** and **force_threshold** are the difference between the two NNP predictions, as the model does not strictly have access to reference data here. That said, the difference between two NNP predictions does have a correlation with the difference between one NNP prediction and a hypothetical DFT calculation at the same point. 

---

As an aside, if you provide the model with too diverse of data (very unphysical configurations), you can get issues where the model is unable to train effectively; trying to explain the very large energy and force difference between a config where atoms are 0.01 A apart and 0.02 A apart is very difficult for a NNP to be frank. Hence in the early stages of  data generation I'd recommend setting the ```maxew``` setting in the lammps interface to something rather small in order to help prevent these unphysical configurations from being generated.

**Comment by Matheus-Novoa** [comment]
I have one more question about how **FDiffMean** and **FDiffMax** parameters are computed. The difference is computed on each component of each atom of each structure, or is it made some kind of normalization?

**Comment by owenrett** [comment]
FDiffMean is the average difference between any two force predictions (from the two NNPs, for a given atom in the structure) for a given structure. FDiffMax is the maximum difference between any two force predictions (from the two NNPs, for a given atom in the structure) for a given structure.

**Comment by Matheus-Novoa** [comment]
Just to clarify, in your example, 12 3 4 are F_x, F_y, F_z, right?
